claim/claimstore_test.go

```diff
@@ -169,8 +226,103 @@ func TestClaimStore_HandlesNotFoundError(t *testing.T) {
 		// inside the error message and not matching it exactly
 		return nil, errors.New("wrapping error message: " + crud.ErrRecordDoesNotExist.Error())
 	}
-	cs := NewClaimStore(mockStore)
+	cs := NewClaimStore(mockStore, nil, nil)
 
-	_, err := cs.Read("missing claim")
+	_, err := cs.ReadClaim("missing claim")
 	assert.EqualError(t, err, ErrClaimNotFound.Error())
 }
+
+func TestStore_EncryptClaims(t *testing.T) {
+	backingStore := crud.NewMockStore()
+	s := NewClaimStore(backingStore, b64encode, b64decode)
+
+	err := s.SaveClaim(exampleClaim)
+	require.NoError(t, err, "SaveClaim failed")
+
+	// Verify that it was encrypted at rest
+	encodedClaimB, err := backingStore.Read(ItemTypeClaims, exampleClaim.ID)
+	require.NoError(t, err, "could not read raw claim data")
+	var gotClaim Claim
+	decodedClaimB, err := b64decode(encodedClaimB)
+	require.NoError(t, err, "failed to decrypt raw claim data")
+	err = json.Unmarshal(decodedClaimB, &gotClaim)
+	require.NoError(t, err, "failed to unmarshal decrypted claim")
+	assert.Equal(t, exampleClaim, gotClaim, "decoded claim doesn't match the original claim")
+
+	// Verify that the claim is decrypted when read
+	gotClaim, err = s.ReadClaim(exampleClaim.ID)
+	require.NoError(t, err, "ReadClaim failed")
+	assert.Equal(t, exampleClaim, gotClaim, "ReadClaim did not round trip the claim properly")
+}
+
+func TestStore_EncryptOutputs(t *testing.T) {
+	writeOnly := func(value bool) *bool {
+		return &value
+	}
+	backingStore := crud.NewMockStore()
+	s := NewClaimStore(backingStore, b64encode, b64decode)
+
+	b := bundle.Bundle{
+		Definitions: map[string]*definition.Schema{
+			"password": {
+				WriteOnly: writeOnly(true),
+			},
+			"port": {
+				WriteOnly: writeOnly(false),
+			},
+		},
+		Outputs: map[string]bundle.Output{
+			"password": {
+				Definition: "password",
+			},
+			"port": {
+				Definition: "port",
+			},
+		},
+	}
+	c, err := New("wordpress", ActionInstall, b, nil)
+	require.NoError(t, err, "New claim failed")
+
+	r, err := c.NewResult(StatusSucceeded)
+	require.NoError(t, err, "NewResult failed")
+
+	err = s.SaveClaim(c)
+	require.NoError(t, err, "SaveClaim failed")
+	err = s.SaveResult(r)
+	require.NoError(t, err, "SaveResult failed")
+
+	password := Output{
+		Claim:  c,
+		Result: r,
+		Name:   "password",
+		Value:  []byte("mypassword"),
+	}
+	err = s.SaveOutput(password)
+	require.NoError(t, err, "SaveOutput failed")
+
+	// Verify that password was encrypted at rest
+	encryptedOutputB, err := backingStore.Read(ItemTypeOutputs, s.outputKey(r.ID, password.Name))
+	require.NoError(t, err, "could not read raw output data")
+	decryptedOutputB, err := b64decode(encryptedOutputB)
+	require.NoError(t, err, "failed to decrypt raw output data")
+	assert.Equal(t, string(password.Value), string(decryptedOutputB), "decrypted output doesn't match the original output")
+
```

Comment:
Do we want to also read directly from the helper (`s.ReadOutput(...)`) here, to confirm the `password` output is returned decrypted as expected?

Reply:
Fixed!